src/Plugins/OutputFormatters/JunitOutputFormatter.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DaveLiddament\StaticAnalysisResultsBaseliner\Plugins\OutputFormatters;
+
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\OutputFormatter\OutputFormatter;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\ResultsParser\AnalysisResults;
+use DOMDocument;
+use Exception;
+use http\Exception\RuntimeException;
+use SimpleXMLElement;
+
+class JunitOutputFormatter implements OutputFormatter
+{
+    /**
+     * @throws Exception
+     */
+    public function outputResults(AnalysisResults $analysisResults): string
+    {
+        if (0 === $analysisResults->getCount()) {
+            return <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites
+        name="SARB" tests="1" failures="0">
+    <testsuite errors="0" tests="1" failures="0" name="Success">
+        <testcase name="Success"/>
+    </testsuite>
+</testsuites>
+
+XML;
+        }
+
+        $xml = $this->getXmlString();
+        $test = new SimpleXMLElement($xml);
+
+        $testCount = (string) $analysisResults->getCount();
+        $test['tests'] = $testCount;
+        $test['failures'] = $testCount;
+
+        $suitCount = 0;
+        $caseCount = 0;
+        $oldRel = null;
+        $testsuite = null;
+        foreach ($analysisResults->getAnalysisResults() as $analysisResult) {
+            $details = $analysisResult->getFullDetails();
+            $type = $details['type'] ?? 'error';
```

Comment:
What is `type` ? Should this be `$analysisResult->getType()->getType()`?

Reply:
Hey @DaveLiddament no, not really. The type we want is the Type that has the value "Warning" or "error", and if it's empty, we thought to just default it to error. Firstly, we used the getType, and it gave another value.
In my understanding, and testing we did `$analysisResult->getType()->getType() !== $details['type']`
Let me know if we unterstood something wrong here.
If you have further questioned we can discuss it of course.